### ISSUE TITLE
Default timeout is set to 24 hours. Should we reduce it?

### ISSUE THREAD

**Issue by nodo** [onsi/ginkgo#507]
We set the default timeout to be 24 hours see [here](https://github.com/onsi/ginkgo/blob/1e72798a2620664d2ab32ffd4473c102fcd4894d/ginkgo/run_watch_and_build_command_flags.go#L154). 

I can see a discussion in https://github.com/onsi/ginkgo/issues/248 but I cannot find context about the value of "24 hours".

It seems to me a very large default and I would like to reduce it? What do you think?

**Comment by tomkukral** [comment]
24h seems to be very very long, something like 30m is sane default for me.

**Comment by onsi** [comment]
24 hours was the longest `go test` would let me set.  With Ginkgo the entire test suite runs within one go `TestX` function and I didn't want integration tests timing out before they were finished (in fact, I didn't want go's timeout to interfere at all) so I maxed out the timeout value.  My thinking was that better test design would have developers being more careful about placing context-appropriate timeout checks on any asynchronous behavior in the tests themselves - and I endeavored to make that pattern easier with Gomega's `Eventually`.  That way developers can get more specific feedback about what exactly is timing out.

At this point I think it would be challenging/surprising to reduce the 24 hour limit.  I'm primarily thinking about very expensive integration test suites out there that may already be taking longer than 30 minutes!
